### ISSUE TITLE
Assigning to !NULL

### ISSUE THREAD

**Issue by fawltylanguage** [gnudatalanguage/gdl#1357]
This method to undefine a variable and free memory does not work in GDL:

```
IDL> a=lindgen(1000000l)
IDL> !null=temporary(a)
IDL> help, a
A               UNDEFINED = <Undefined>
IDL> 

GDL> a=lindgen(1000000l)
GDL> !null=temporary(a)
% NullGDL::EqType(...) called.
% Execution halted at: $MAIN$
GDL> 
```

**Comment by GillesDuvert** [comment]
hopefully closed by #1644

**Comment by GillesDuvert** [comment]
closed by https://github.com/gnudatalanguage/gdl/pull/1644
